Source/WebCore/Modules/url-pattern/URLPattern.cpp

```diff
@@ -26,29 +26,239 @@
+            && init.username.isNull())
+            result.username = processBaseURLString(baseURL.user(), type);
+
+        if (type != BaseURLStringType::Pattern
+            && init.protocol.isNull()
+            && init.hostname.isNull()
+            && init.port.isNull()
+            && init.username.isNull()
+            && init.password.isNull())
+            result.password = processBaseURLString(baseURL.password(), type);
+
+        if (init.protocol.isNull()
+            && init.hostname.isNull()) {
+            String baseHost = (!baseURL.host().isNull()) ? baseURL.host().toString() : ""_s;
+            result.hostname = processBaseURLString(baseHost, type);
+        }
+
+        if (init.protocol.isNull()
+            && init.hostname.isNull()
+            && init.port.isNull()) {
+            auto port = baseURL.port();
+            result.port = port ? String::number(*port) : ""_s;
+        }
+
+        if (init.protocol.isNull()
+            && init.hostname.isNull()
+            && init.port.isNull()
+            && init.pathname.isNull()) {
+            String serializedBaseURL = baseURL.path().toString();
+            result.pathname = processBaseURLString(serializedBaseURL, type);
+        }
+
+        if (init.protocol.isNull()
+            && init.hostname.isNull()
+            && init.port.isNull()
+            && init.pathname.isNull()
+            && init.search.isNull()) {
+            String urlQuery = baseURL.hasQuery() ? baseURL.query().toString() : ""_s;
+            result.search = processBaseURLString(urlQuery, type);
+        }
+
+        if (init.protocol.isNull()
+            && init.hostname.isNull()
+            && init.port.isNull()
+            && init.pathname.isNull()
+            && init.search.isNull()
+            && init.hash.isNull()) {
+            String urlFragment = baseURL.hasFragmentIdentifier() ? baseURL.fragmentIdentifier().toString() : ""_s;
+            result.hash = processBaseURLString(urlFragment, type);
+        }
+    }
+
+    if (!init.protocol.isNull()) {
+        ExceptionOr<String> protocolResult = canonicalizeProtocol(WTFMove(init.protocol), type);
+
+        if (protocolResult.hasException())
+            return protocolResult.releaseException();
+
+        result.protocol = protocolResult.releaseReturnValue();
+    }
+
+    if (!init.username.isNull())
+        result.username = canonicalizeUsername(WTFMove(init.username), type);
+
+    if (!init.password.isNull())
+        result.password = canonicalizePassword(WTFMove(init.password), type);
+
+    if (!init.hostname.isNull()) {
+        ExceptionOr<String> hostResult = canonicalizeHost(WTFMove(init.protocol), type);
+
+        if (hostResult.hasException())
+            return hostResult.releaseException();
+
+        result.hostname = hostResult.releaseReturnValue();
+    }
+
+    if (!init.port.isNull()) {
+        ExceptionOr<String> portResult = canonicalizePort(WTFMove(init.port), WTFMove(init.protocol), type);
```

Comment:
`auto`

Reply:
We need to be careful with all these `WTFMove()` calls, at line 176, you already move `init.protocol` so I think we're doing a potential use-after-move here. I wouldn't be surprised if we had the same issue in other places. My advice would be to stop using `WTMove()` on `init` values.